pkg/operator/controllers/alertwebhook/alertwebhook_controller.go

```diff
@@ -35,21 +34,22 @@ var alertManagerName = types.NamespacedName{Name: "alertmanager-main", Namespace
 type Reconciler struct {
 	log *logrus.Entry
 
-	arocli        aroclient.Interface
 	kubernetescli kubernetes.Interface
+
+	client client.Client
 }
 
-func NewReconciler(log *logrus.Entry, arocli aroclient.Interface, kubernetescli kubernetes.Interface) *Reconciler {
+func NewReconciler(log *logrus.Entry, kubernetescli kubernetes.Interface) *Reconciler {
 	return &Reconciler{
 		log:           log,
-		arocli:        arocli,
 		kubernetescli: kubernetescli,
 	}
 }
 
 // Reconcile makes sure that the Alertmanager default webhook is set.
 func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
-	instance, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	instance := &arov1alpha1.Cluster{}
+	err := r.client.Get(ctx, types.NamespacedName{Name: arov1alpha1.SingletonClusterName}, instance)
```

Comment:
Since we can't guarantee at the time of construction in of any of these controllers its ability to get the cluster, should we be introducing a guard statement around if `r.client` was populated before attempting to utilize it?

Reply:
`InjectClient` will be called by the builder (down the `builder.Named(ControllerName).Complete(r)` call stack). So at the moment of reconciliation there will always be a client here: 

And this is how the docs suggests to set the client in this [example](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.1/pkg/builder#example-Builder).

However thi is very implicit and I prefer being explicit. I think there might be a way to explicitly pass the client into the constructor. I was thinking - `ctrl.Manager` should already have an instance of a client and we create it before calling any of the constructors, so we should be able `mgr.GetClient()` and pass into the constructor. But I didn't have time to dig into this and I don't fully understand implications so I decided to follow an example from the docs for now.

Reply:
Thank you for sharing this, I agree to start we should follow the example and enhance as we see fit down the line.